### DISCUSSION TITLE
[BUG/Help] 如何训练逻辑推理能力

### DISCUSSION THREAD

**Discussion by netwolf712** [zai-org/ChatGLM-6B#1379] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

如题：
目前使用微调功能可以教会他专业领域的信息提取能力，但是碰到一些需要简单换算的就不知道怎么教了。
比如给定一个指标，此指标在[1,10]范围内的都是正常，其余为异常。

应该如何微调使得他认识这个区间，当下次碰到这个指标时，自动根据指标数值判断是否异常。

### Steps To Reproduce

无

### Environment

```markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :
```

**Comment by cywjava** [comment]
这个就高深了

**Comment by cywjava** [comment]
之前有人说过，比如问他10以内的加减法，他能给出正确答案，但是换大一点的，就不行了，怀疑是它记住了这些而已，根本没学会加减法，只是因为以前它学过这些内容，

**Comment by Dragonkingpan** [comment]
不是那样的，处理数学问题，你先让他学会编程，然后再让他用程序处理，比如：
问：
15615+54586=
答：
15615 + 54586 = 70201
问：
你是怎么做出来的
答：
我是通过程序计算得出的。在程序中，我将这两个数相加并输出结果70201。
问：
程序怎么计算
答：
在程序中，我使用了加法运算符“+”来计算这两个数的和。具体来说，我将15615和54586分别赋值给两个变量，然后使用加法运算符将它们相加，最终得到70201这个结果。以下是Python程序的示例代码：
num1 = 15615
num2 = 54586
result = num1 + num2
print("结果为：", result)
程序运行后，输出结果为70201。
